client/src/modules/dashboard/Events/pages/EditEventPage.tsx

```diff
@@ -6,63 +6,55 @@ import { IEventFormData } from '../components/EventFormUtils';
 import Layout from '../../shared/components/Layout';
 import Skeleton from '../../Venues/components/Skeleton';
 import EventForm from '../components/EventForm';
-import { useCreateEventMutation, useEventQuery } from '../../../../generated';
+import { useEventQuery, useUpdateEventMutation } from '../../../../generated';
 import { EVENTS } from '../graphql/queries';
 import { getId } from '../../../../helpers/getId';
 
 export const EditEventPage: NextPage = () => {
   const router = useRouter();
-  const [loading, setLoading] = useState<boolean>(false);
+  const [loadingUpdate, setLoadingUpdate] = useState<boolean>(false);
   const id = getId(router.query) || -1;
 
   const { loading: eventLoading, error, data } = useEventQuery({
     variables: { id },
   });
 
-  const [createEvent] = useCreateEventMutation({
+  const [updateEvent] = useUpdateEventMutation({
     refetchQueries: [{ query: EVENTS }],
   });
 
   const onSubmit = async (data: IEventFormData) => {
     // TODO: load chapter from url or something like that
-    setLoading(true);
+    setLoadingUpdate(true);
 
     try {
-      const HARD_CODE = { chapterId: 1 };
-
-      console.log(data.start_at);
-
       const eventData = {
         ...data,
         capacity: parseInt(String(data.capacity)),
         start_at: new Date(data.start_at).toISOString(),
         ends_at: new Date(data.ends_at).toISOString(),
-        ...HARD_CODE,
         tags: undefined,
       };
 
-      const event = await createEvent({
-        variables: { data: { ...eventData } },
+      const event = await updateEvent({
+        variables: { id, data: { ...eventData } },
       });
 
       if (event.data) {
-        router.replace(
-          `/dashboard/events/[id]`,
-          `/dashboard/events/${event.data.createEvent.id}`,
-        );
+        await router.replace('/dashboard/events');
```

Comment:
For updates I'd do a push instead of a replace, because going back will be a valid option, compared to creatingEvents where when you go back you'd reach an empty form

Reply:
got it, I'll update venues too